src/modules/75-cf/components/FlagActivation/LinkJiraIssueModal.tsx

```diff
@@ -0,0 +1,190 @@
+  SelectOption,
+  ButtonSize,
+  FormInput,
+  Formik,
+  FormikForm,
+  useToaster
+} from '@harness/uicore'
+
+import { useParams } from 'react-router-dom'
+import * as yup from 'yup'
+import { getErrorMessage } from '@cf/utils/CFUtils'
+import { useStrings } from 'framework/strings'
+import RbacButton from '@rbac/components/Button/Button'
+import { ResourceType } from '@rbac/interfaces/ResourceType'
+import { PermissionIdentifier } from '@rbac/interfaces/PermissionIdentifier'
+import useActiveEnvironment from '@cf/hooks/useActiveEnvironment'
+import { useGetJiraIssues, usePatchFeature } from 'services/cf'
+import { FeatureIdentifier } from 'framework/featureStore/FeatureIdentifier'
+import patch from '../../utils/instructions'
+
+export interface LinkJiraIssueModalProps {
+  featureIdentifier: string
+  refetchFlag: () => void
+}
+
+const LinkJiraIssueModal: FC<LinkJiraIssueModalProps> = ({ featureIdentifier, refetchFlag }) => {
+  const { showError, showSuccess } = useToaster()
+
+  const { accountId: accountIdentifier, orgIdentifier, projectIdentifier } = useParams<Record<string, string>>()
+  const { getString } = useStrings()
+  const { activeEnvironment } = useActiveEnvironment()
+
+  const [isModalDisplayed, setIsModelDisplayed] = useState<boolean>(false)
+  const hideModal = useCallback(() => setIsModelDisplayed(false), [])
+  const openModal = useCallback(() => setIsModelDisplayed(true), [])
+
+  const queryParams = {
+    projectIdentifier,
+    environmentIdentifier: activeEnvironment,
+    accountIdentifier,
+    orgIdentifier
+  }
+
+  const { mutate, loading: patchLoading } = usePatchFeature({
+    identifier: featureIdentifier,
+    queryParams
+  })
+
+  const {
+    data,
+    loading: jiraIssuesLoading,
+    refetch
+  } = useGetJiraIssues({
+    lazy: true,
+    debounce: 700
+  })
+
+  const jiraItems = useMemo<SelectOption[]>(() => {
+    if (data?.issues?.length) {
+      return data.issues.map(issue => ({ label: issue.key as string, value: issue.key as string }))
+    }
+    return []
+  }, [data?.issues])
+
+  const handleSubmit = (jiraIssueKey: string): void => {
+    patch.feature.addInstruction({
+      kind: 'addJiraIssueToFlag',
+      parameters: {
+        issueKey: jiraIssueKey as string
+      }
+    })
+
+    patch.feature.onPatchAvailable(async instructions => {
+      try {
+        await mutate(instructions)
+        hideModal()
+        refetchFlag()
+        showSuccess(getString('cf.featureFlags.jira.successMessage'))
+      } catch (e) {
+        showError(getErrorMessage(e), undefined, 'cf.featureFlags.jira.errorMessage')
+      } finally {
+        patch.feature.reset()
+      }
+    })
+  }
+
+  return (
+    <>
+      <Layout.Horizontal flex={{ justifyContent: 'flex-start' }}>
+        <RbacButton
+          data-testid="add-jira-issue-modal"
+          minimal
+          icon="small-plus"
+          size={ButtonSize.SMALL}
+          font={{ size: 'small' }}
+          style={{ padding: 0 }}
```

Comment:
Could we use the variation type of `LINK` instead?
```suggestion
          variation={ButtonVariation.LINK}
```